### ISSUE TITLE
chefignore requires double definition

### ISSUE THREAD

**Issue by fgarcia** [chef/chef#2060]
As of Chef 11.16 I must add twice each file I want to ignore (at least in my cookbooks/chefignore file), like this:

```
# Ignore in cookbook subfolders
*/fooo.txt 

# ignore in cookbook root 
fooo.txt 
```

The way to reproduce this was:

```
 knife cookbook upload my_cookbook
 knife cookbook show my_cookbook | grep fooo.txt
```

Since I expected something similar to the Git syntax, I was surprised by this. Is that the expected behavior?

**Comment by rychipman** [comment]
It would be possible to attempt to mimic gitignore globbing, but it would be nontrivial (I checked, and this can't be fixed by setting FNM_XXX flags in File.fnmatch?), and could just introduce other bugs/confusion.

Is this only as of 11.16? Was there a point in the past where the normal gitignore syntax worked? Because looking at the code, this is just how ruby's File.fnmatch? works, and it seems to have been implemented the same way as far back as 10-stable. Additionally, I believe (though this is mostly anecdotal) that gitignore's globbing is actually the exception to the rule -- I have found that this fnmatch behavior appears to be implemented in most other instances of globbing. Gulp is one example that comes to mind.

Just my 2 cents, but it seems like sticking with the default behavior would be more consistent and less confusing than changing it.
